cpp/src/arrow/util/io_util.cc

```diff
@@ -230,6 +230,150 @@ class ErrnoDetail : public StatusDetail {
+    case ERROR_FILENAME_EXCED_RANGE:  //  206
+      return ENOENT;
+
+    case ERROR_BAD_ENVIRONMENT:  //   10
+      return E2BIG;
+
+    case ERROR_BAD_FORMAT:                 //   11
+    case ERROR_INVALID_STARTING_CODESEG:   //  188
+    case ERROR_INVALID_STACKSEG:           //  189
+    case ERROR_INVALID_MODULETYPE:         //  190
+    case ERROR_INVALID_EXE_SIGNATURE:      //  191
+    case ERROR_EXE_MARKED_INVALID:         //  192
+    case ERROR_BAD_EXE_FORMAT:             //  193
+    case ERROR_ITERATED_DATA_EXCEEDS_64k:  //  194
+    case ERROR_INVALID_MINALLOCSIZE:       //  195
+    case ERROR_DYNLINK_FROM_INVALID_RING:  //  196
+    case ERROR_IOPL_NOT_ENABLED:           //  197
+    case ERROR_INVALID_SEGDPL:             //  198
+    case ERROR_AUTODATASEG_EXCEEDS_64k:    //  199
+    case ERROR_RING2SEG_MUST_BE_MOVABLE:   //  200
+    case ERROR_RELOC_CHAIN_XEEDS_SEGLIM:   //  201
+    case ERROR_INFLOOP_IN_RELOC_CHAIN:     //  202
+      return ENOEXEC;
+
+    case ERROR_INVALID_HANDLE:         //    6
+    case ERROR_INVALID_TARGET_HANDLE:  //  114
+    case ERROR_DIRECT_ACCESS_HANDLE:   //  130
+      return EBADF;
+
+    case ERROR_WAIT_NO_CHILDREN:    //  128
+    case ERROR_CHILD_NOT_COMPLETE:  //  129
+      return ECHILD;
+
+    case ERROR_NO_PROC_SLOTS:        //   89
+    case ERROR_MAX_THRDS_REACHED:    //  164
+    case ERROR_NESTING_NOT_ALLOWED:  //  215
+      return EAGAIN;
+
+    case ERROR_ARENA_TRASHED:      //    7
+    case ERROR_NOT_ENOUGH_MEMORY:  //    8
+    case ERROR_INVALID_BLOCK:      //    9
+    case ERROR_NOT_ENOUGH_QUOTA:   // 1816
+      return ENOMEM;
+
+    case ERROR_ACCESS_DENIED:            //    5
+    case ERROR_CURRENT_DIRECTORY:        //   16
+    case ERROR_WRITE_PROTECT:            //   19
+    case ERROR_BAD_UNIT:                 //   20
+    case ERROR_NOT_READY:                //   21
+    case ERROR_BAD_COMMAND:              //   22
+    case ERROR_CRC:                      //   23
+    case ERROR_BAD_LENGTH:               //   24
+    case ERROR_SEEK:                     //   25
+    case ERROR_NOT_DOS_DISK:             //   26
+    case ERROR_SECTOR_NOT_FOUND:         //   27
+    case ERROR_OUT_OF_PAPER:             //   28
+    case ERROR_WRITE_FAULT:              //   29
+    case ERROR_READ_FAULT:               //   30
+    case ERROR_GEN_FAILURE:              //   31
+    case ERROR_SHARING_VIOLATION:        //   32
+    case ERROR_LOCK_VIOLATION:           //   33
+    case ERROR_WRONG_DISK:               //   34
+    case ERROR_SHARING_BUFFER_EXCEEDED:  //   36
+    case ERROR_NETWORK_ACCESS_DENIED:    //   65
+    case ERROR_CANNOT_MAKE:              //   82
+    case ERROR_FAIL_I24:                 //   83
+    case ERROR_DRIVE_LOCKED:             //  108
+    case ERROR_SEEK_ON_DEVICE:           //  132
+    case ERROR_NOT_LOCKED:               //  158
+    case ERROR_LOCK_FAILED:              //  167
+    case 35:                             //   35 (undefined)
```

Comment:
I suppose this is somewhat "vendored" code but why is "35 (undefined)" fall into the category EACCES?

Reply:
The legacy version of that code was generated by calling the `_dosmaperr` function, which apparently isn't supported anymore. That function did return a mapping for the Windows error code 35, which doesn't have a corresponding macro.